data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/index/IndexManager.java

```diff
@@ -34,14 +43,100 @@
     protected RestHighLevelClient restHighLevelClient;
     protected OpenSearchSinkConfiguration openSearchSinkConfiguration;
     protected IsmPolicyManagementStrategy ismPolicyManagementStrategy;
+    protected String indexPrefix;
 
     private static final Logger LOG = LoggerFactory.getLogger(IndexManager.class);
 
+    //For matching a string that begins with a "%{" and ends with a "}".
+    //For a string like "data-prepper-%{yyyy-MM-dd}", "%{yyyy-MM-dd}" is matched.
+    private final static String TIME_PATTERN_REGULAR_EXPRESSION  = "%\\{.*?\\}";
```

Comment:
Consider revising pattern to reduce additional validation. `"%\\{([A-Za-z-#\\{}\\[\\]']*)}$"` matches all valid DateTimeFormatter characters at the end of a string surrounded by `%{}`

Reply:
My thought is we should keep the index name flexible. The pattern is part of the index name.  Inside `%{}`, there could be something like "-.letterYYYY", etc. We still want to support strings other than Java DataTimeFormatter symbols.  

We know OpenSearch index special characters are similar to ElasticSearch special character set: https://stackoverflow.com/questions/41585392/what-are-the-rules-for-index-names-in-elastic-search

It's more feasible for us to list out the characters we don't support. It is hard to list out all the characters that are allowed to be between %{}

Reply:
Good idea allowing for flexible names.